driver-core/src/main/com/mongodb/ReadConcern.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2015 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb;
+
+import org.bson.BsonDocument;
+import org.bson.BsonString;
+
+import static com.mongodb.assertions.Assertions.notNull;
+
+/**
+ * The readConcern option allows clients to choose a level of isolation for their reads.
+ *
+ * The WiredTiger storage engine, introduces the readConcern option for replica sets and replica set shards.  Allowing clients to choose a
+ * level of isolation for their reads.
+ *
+ * @mongodb.server.release 3.2
+ * @since 3.2
+ */
+public final class ReadConcern {
```

Comment:
While closer to the C# design, there are still a few differences to resolve resolve, as i don't see a reason for them to be different.

# In C# driver, the readConcernLevel property itself is Optional.  A null value indicates the default level
# In C# driver, there is no ReadConcernLevel.DEFAULT.  

I think the C# design is closer to what the wire protocol actually allows: there is no way to specify a read concern level corresponding to default, e.g.

   readConcern { level : ???? }

But there is a concept of the default read concern, which is where nothing at all is specified and the readConcern document is excluded.